src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2681,6 +2681,26 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         return nullptr;
     }
 
+    ASR::asr_t* handle_intrinsic_len(Allocator &al, Vec<ASR::call_arg_t> args,
+                                        const Location &loc) {
+        ASR::expr_t *arg = args[0].m_value;
+        ASR::ttype_t *type = ASRUtils::expr_type(arg);
+        ASR::ttype_t *to_type = ASRUtils::TYPE(ASR::make_Integer_t(al, loc,
+                                    4, nullptr, 0));
+        if (ASRUtils::is_character(*type)) {
```

Comment:
In here you check if `arg` is a `StringConstant` and if so, set `value` to `std::string(down_cast<ASR::StringConstant_t>(arg)->m_s).size()` (converted to `IntegerConstant`).